### DISCUSSION TITLE
Fit several fibre in one ODF

### DISCUSSION THREAD

**Discussion by CharbelSakr26** [mtex-toolbox/mtex#1724] [Ask Anything]
Hello,
Is it possible to fit an ODF composed of different fibers and get the fit for each fiber individually? I know the fit function returns only one fiber, but what if we have two or more fibers in one ODF.
As an example: 
![Capture](https://user-images.githubusercontent.com/118815152/214316467-11c30c96-522a-44ae-93b0-9a76a483ee4a.PNG)
Here we have two types of fibre: 1: fiber with h(0,0,1) | | Z and 2: fiber with h(1,0,0) | | X. 
I would like after that to get The 'volume' of an orientation that is close to one of each fiber. 
Thank you!
Charbel.

**Comment by kilir** [comment]
Hi,
do you would like to do that for discrete orientations or is it that you'd have an odf and you'd like to decompose it into fibre components?

Cheers,
Rüdiger

**Reply by CharbelSakr26** [reply]
Dear Mr Rüdiger,
Thank you for your reply.
Yes, I would like to decompose an ODF into several fiber components like in the one that I showed. I would like to use the fibre.fit function on ODFs that are composed of different type of fibres and eventually get all the fibres type individually from one ODF.  Is that possible? 
Thank you!
Charbel.

**Reply by kilir** [reply]
fibre.fit uses orientations as an input, not an odf and so far, calcComponents does not find fibreComponents.
One option which may work in certain cases  -  in other cases it will absolutely not work - is to
a) sample orientations from the odf then b) do a clustering on those and then use fibre.fit on the orientations of each cluster.
```
ori = odf.discreteSample(1000)
[cId,center] = calcCluster(ori,'numCluster',2,'method','hierarchical');
f1 = fibre.fit(ori(cId==1))
f2 = fibre.fit(ori(cId==2))
```

However, chances are quite high it may only work for very specific sets of fibres.

Cheers,
Rüdiger

**Comment by Claeror** [comment]
Hello,

I faced the same problem and the solution i came up with was to assume one of the fibres to be approximately known, and then divide the orientations data based on that fibre. The supposed fibre was found "by hand" by trying orientations and common fibres (alpha, beta etc...) but it is only used for separation purpose.
```
assumedFibre = fibre(...)
data1 = dataFull( angle(assumedFibre ,dataFull.orientations) > 20*degree) 
data2 = dataFull( angle(assumedFibre ,dataFull.orientations) < 20*degree)
fibre1 = fibre.fit(data1)
fibre2 = fibre.fit(data2)
```
The 20° angle is only based on visual criterion for a good separation of the fibres on the plot3d and can differ in your case

It's far from being a perfect solution but it work pretty well if you can find a close enough fibre "by hand"

Cheers,
Romain.

PS :  you can check if your assumedFibre is far from the fit fibre with the angle fonciton between both
